components/search_results/search_results.tsx

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+/* eslint-disable react/no-string-refs */
+
+import React, {useEffect, useRef} from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+
+import classNames from 'classnames';
+
+import {debounce} from 'mattermost-redux/actions/helpers';
+
+import * as Utils from 'utils/utils.jsx';
+import {searchHintOptions} from 'utils/constants';
+
+import SearchResultsHeader from 'components/search_results_header';
+import SearchResultsItem from 'components/search_results_item';
+import SearchHint from 'components/search_hint/search_hint';
+import LoadingSpinner from 'components/widgets/loading/loading_wrapper';
+import NoResultsIndicator from 'components/no_results_indicator/no_results_indicator';
+import FlagIcon from 'components/widgets/icons/flag_icon';
+
+import {NoResultsVariant} from 'components/no_results_indicator/types';
+
+import type {Props} from './types';
+
+const GET_MORE_BUFFER = 30;
+
+const renderView = (props: Record<string, unknown>): JSX.Element => (
+    <div
+        {...props}
+        className='scrollbar--view'
+    />
+);
+
+const renderThumbHorizontal = (props: Record<string, unknown>): JSX.Element => (
+    <div
+        {...props}
+        className='scrollbar--horizontal'
+    />
+);
+
+const renderThumbVertical = (props: Record<string, unknown>): JSX.Element => (
+    <div
+        {...props}
+        className='scrollbar--vertical'
+    />
+);
+
+export const shouldRenderFromProps = (props: Props, nextProps: Props): boolean => {
```

Comment:
Nothing to change right now, but we may not actually need this going forward. The previous `SearchResults` was written before we were using `PureComponent` everywhere, and I don't think we understood `shouldComponentUpdate` at the time either